rocAL/rocAL/source/master_graph.cpp

```diff
@@ -848,17 +906,44 @@ void MasterGraph::output_routine()
             // Multiple cycles worth of internal_batch_size images should be processed to complete a full _user_batch_size
             for(unsigned cycle_idx = 0; cycle_idx< _user_to_internal_batch_ratio; cycle_idx++)
             {
-                // Swap handles on the input image, so that new image is loaded to be processed
-                auto load_ret = _loader_module->load_next();
-                if (load_ret != LoaderModuleStatus::OK)
-                    THROW("Loader module failed to load next batch of images, status " + TOSTR(load_ret))
+                if(is_video_loader())
+                {
+                    auto load_ret = _video_loader_module->load_next();
+                    if (load_ret != VideoLoaderModuleStatus::OK)
+                        THROW("Video Loader module failed to load next batch of images, status " + TOSTR(load_ret))
+                }
+                else
+                {
+                     // Swap handles on the input image, so that new image is loaded to be processed
+                    auto load_ret = _loader_module->load_next();
+                    if (load_ret != LoaderModuleStatus::OK)
+                        THROW("Loader module failed to load next batch of images, status " + TOSTR(load_ret))
+                }
+
 
                 if (!_processing)
                     break;
-                auto this_cycle_names =  _loader_module->get_id();
-                auto decode_image_info = _loader_module->get_decode_image_info();  
-                auto crop_image_info = _loader_module->get_crop_image_info();              
 
+                decoded_image_info decode_image_info;
+                std::vector<std::string> this_cycle_names;
+                crop_image_info crop_image_info;
+                std::vector<size_t> sequence_start_framenum;
+                std::vector<std::vector<float>> sequence_frame_timestamp;
+                if(is_video_loader())
+                {
+                    this_cycle_names = _video_loader_module->get_id();
+                    decode_image_info = _video_loader_module->get_decode_image_info();
+                    sequence_start_framenum = _video_loader_module->get_sequence_start_frame_number();
+                    sequence_frame_timestamp = _video_loader_module->get_sequence_frame_timestamps();
+                    full_batch_sequence_start_frame_number += sequence_start_framenum;
```

Comment:
please check this line. full_batch_sequence_start_frame_number is an empty vector and adding an offset to uninitialized vector can cause memory access violation. True for full_batch_sequence_frame_timestamps

Reply:
We have overloaded the operator to append vector and vector of vectors. We have used the standard insert function in the operator overloading so there won't be any memory issues.